cl/corpus_importer/management/commands/financial_disclosures.py

```diff
@@ -0,0 +1,344 @@
+        logger.info(
+            f"Preparing to process JW url: {quote(data['url'], safe=':/')}"
+        )
+
+        pdf_response = requests.get(data["url"], timeout=60 * 20)
+
+    elif data["disclosure_type"] == "single":
+        # Split single long tiff into multiple tiffs and combine into PDF
+        logger.info(
+            f"Preparing to process url: {quote(data['url'], safe=':/')}"
+        )
+        pdf_response = requests.post(
+            settings.BTE_URLS["image-to-pdf"],
+            params={"tiff_url": data["url"]},
+            timeout=10 * 60,
+        )
+    else:
+        # Combine split tiffs into one single PDF
+        logger.info(
+            f"Preparing to process split urls: "
+            f"{quote(data['urls'][0], safe=':/')}"
+        )
+
+        pdf_response = requests.post(
+            settings.BTE_URLS["urls-to-pdf"],
+            json=json.dumps({"urls": data["urls"]}),
+        )
+    return pdf_response
+
+
+def import_financial_disclosures(options):
+    """Import financial documents into www.courtlistener.com
+
+    :param options: argparse
+    :return:None
+    """
+    filepath = options["filepath"]
+    with open(filepath) as f:
+        disclosures = json.load(f)
+
+    for data in disclosures:
+        # Generate PDF content from our three paths
+        if options["skip_until"]:
+            if data["id"] < int(options["skip_until"]):
+                continue
+
+        year = data["year"]
+        person_id = data["person_id"]
+        logger.info(
+            f"Processing id:{person_id} " f"year:{year}, with id:{data['id']}"
+        )
+
+        pdf_response = extract_pdf(data)
+        pdf_bytes = pdf_response.content
+
+        if pdf_response.status_code != 200:
+            logger.info("PDF generation failed.")
+            continue
+
+        logger.info("PDF content generated. Extracting content now.")
+
+        # Sha1 hash - Check for duplicates
+        sha1_hash = sha1(pdf_bytes)
+        in_system = check_if_in_system(sha1_hash)
+        if in_system:
+            logger.info("\x1b[6;30;41m" + "PDF already in system." + "\x1b[0m")
+            continue
+
+        # Return page count - 0 indicates a failure of some kind.  Like PDF
+        # Not actually present on aws.
+        pg_count = get_page_count(pdf_bytes)
+        if pg_count == 0:
+            logger.info("\x1b[6;30;41m" + "PDF failed!" + "\x1b[0m")
+            return
+
+        # Save Financial Disclosure here to AWS and move onward
+        disclosure = FinancialDisclosure(
+            year=year,
+            page_count=pg_count,
+            person=Person.objects.get(id=person_id),
+            pdf_hash=sha1_hash,
+            has_been_extracted=False,
+        )
+        # Save and upload PDF
+        disclosure.filepath.save("", ContentFile(pdf_bytes))
+        logger.info(
+            f"Uploaded to https://{settings.AWS_S3_CUSTOM_DOMAIN}/"
+            f"{disclosure.filepath}"
+        )
```

Comment:
Shouldn't this be part of the transaction?

Reply:
My thinking was that - there is usefulness in having the PDF generated and uploaded to AWS - if something fails -we know it failed from the database.  thus we dont need to pull down the large TIFF files instead we can pull our newly created smaller PDFs